src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiParameterGenerator.cs

```diff
@@ -128,8 +127,8 @@ public static IList<OpenApiParameter> CreateParameters(this ODataContext context
                     parameter = new OpenApiParameter
                     {
                         Name = parameterNameMapping == null ? edmParameter.Name : parameterNameMapping[edmParameter.Name],
-                        In = ParameterLocation.Path,
-                        Required = true,
+                        In = edmParameter is IEdmOptionalParameter ? ParameterLocation.Query : ParameterLocation.Path,
+                        Required = edmParameter is not IEdmOptionalParameter,
```

Comment:
as one is the inverse of the other, just assign the result to a boolean var above, and use that boolean instead in both places